Feat/Scrum 107 이용약관 UI 구현

inwoodev opened the pull request:
## PR 타입
어떤 변경에 대한 PR인가요?

<!-- 아래 체크리스트 중 해당 되는 부분에 체크해주세요 "x". -->

- [x] 버그 수정
- [x] Feature / 신기능
- [ ] 코드 스타일 업데이트 (formatting, local variables)
- [ ] 리팩토링 (no functional changes, no api changes)
- [ ] 빌드 관련 변경사항
- [ ] CI 관련된 변경사항
- [ ] Documentation content changes
- [ ] 구조 변경
- [ ] Other... 추가 설명 요망:


## 배경
온보딩 기능을 구현함에 있어서 유저가 서비스 가입 하기 전 약관을 동의할 수 있는 화면이 필요합니다.

## 목표
- 피그마 디자인과 일치하는 약관 화면 UI 구현

## 결과 (Optional)

![Simulator Screen Recording - iPhone 15 - 2024-06-09 at 22 22 53](https://github.com/project-lyrics/app-iOS/assets/69072471/bc1acad9-9d90-402a-a878-f2e7eae0b2e4)


## Trouble or Trouble Shooting(Optional)

<img width="1288" alt="Screenshot 2024-06-09 at 9 18 48 PM" src="https://github.com/project-lyrics/app-iOS/assets/69072471/5aa9f45a-760b-47ad-bab7-e69df2138cef">

피그마에는 전체동의 버튼의 백그라운드 색상이 pressedBrand라고 적혀있지만 실제 적용시 색상이 의도된 대로 보여지지 않는 문제가 있어서

일단은 `alertSuccess` 색상으로 대체하였습니다.


## 다크모드와 외부 모듈 문제
@derrickkim0109  @noeyiz 
외부 모듈에 존재하는 Color asset의 다크모드 설정이 적용이 되지 않는 문제가 있습니다.

SharedDesignSystem에 등록되어있는 Color Asset들의 dark appearance가 현재 적용이 되지 않는 문제가 있습니다.
이게 정말 외부 모듈에 있는 컬러 에셋의 다크모드 설정이 안되는건지 아니면 해결 방법이 있는지는 추후 같이 리서치 해 보면 좋을 듯 합니다.

관련 내용을 리서치 해 보았고 스택오버플로우에서 다음과 같은 내용을 확인하였습니다.
> Beware of using color assets from other modules as well:
While they appear to work, it is only because when you assign color assets in a storyboard, Xcode actually encodes the color information from the asset in the storyboard file itself. One sign this is not intended to work is that if you define a Color Set asset with a variation for Dark Appearance, only the Any Appearance color will be used (even in dark mode). Color Set assets in the same module, however, work normally.
[관련 링크](https://stackoverflow.com/questions/59452660/how-to-use-image-contained-in-resource-framework-from-main-module)

inwoodev committed:
design: CheckBox 이미지 에셋 수정

다크모드, 라이트모드를 별도의 에셋으로 나누지 않고 하나의 에셋으로 통합

inwoodev committed:
design: 피그마에 명시된 색상 네임과 다른 컬러 네임 수정

inwoodev committed:
feat: OnboardingExample 환경설정

- SceneDelegate, AppDelegate 및 Tuist세팅을 통해서 샘플앱을 실행할 수 있는 환경 설정
- exampleAppDefaultSettings에서 -all_load 세팅을 제거하여 duplicate symbol 문제 해결

inwoodev committed:
feat: CheckBoxButton 뷰 구현 & 추가

- selected & unselected 상태에 따라서 체크박스 색상을 변경합니다.
- 생성시 View를 주입할 수 있고 해당 버튼 내부에 보여줄 수 있도록 레이아웃 설정이 가능합니다.

inwoodev committed:
feat: UserAgreementListView 구현 & 추가

- 전체 동의 및 세부 동의 버튼을 포함하는 ListView
- 전체동의 버튼을 탭할 경우 화면의 백그라운드 색상 및 체크 버튼이 활성화된다.
- 그 외 세부 버튼을 텝할 경우 체크 버튼이 활성화 된다.

inwoodev committed:
feat<공용컴포넌트> FeelinConfirmButton 제작

- 비활성화시 회색, 활성화시 active 색상으로 변경되는 버튼 구현
- 필요시 버튼 타이틀, 그리고 활성화 상태 뿐 아니라 각 상태에 따른 색상을 외부에서 지정할 수 있도록 세팅

inwoodev committed:
feat: UseAgreementViewController UI 구현

- 약관 타이틀뷰를 상단에 표시
- 이용약관 버튼을 모두 확인 가능
- 시작하기 버튼은 디폴트로 비활성화 하여 화면 맨 하단에 배치

inwoodev committed:
move: OnboardingView & RootViewController 파일 이동

inwoodev commented on Plugins/DependencyPlugin/ProjectDescriptionHelpers/Project+Environment.swift:
```diff
@@ -45,7 +45,6 @@ public extension Project {
         )
         public static let exampleAppDefaultSettings: Settings = .settings(
             base: [
-                "OTHER_LDFLAGS": "$(inherited) -Xlinker -interposable -all_load",
                 "GCC_PREPROCESSOR_DEFINITIONS": "$(inherited) FLEXLAYOUT_SWIFT_PACKAGE=1"
             ],
             configurations: [
```
app에 해당 설정을 추가할 경우 duplicatesymbol 에러가 발생하며 데모앱을 실행할 수 없습니다.

inwoodev commented on Projects/Feature/Onboarding/Interface/Sources/Components/CheckBoxButton.swift:
```diff
@@ -0,0 +1,104 @@
+        self.checkBoxImageView.image = image
+    }
+    
+    private func setUpView() {
+        flexContainer.isUserInteractionEnabled = false
+        
+        flexContainer.flex
+            .alignSelf(.center)
+            .direction(.row).define { flex in
+            flex.addItem(checkBoxImageView)
+                .marginLeft(18)
+            
+            if let additionalView = additionalView {
+                flex.addItem(additionalView)
+            }
+        }
+    }
```
버튼 내부에 flexLayout의 flexContainer를 사용할 경우 `isUserInteractionEnabled`설정을 하지 않을 경우 버튼이 부분적으로 탭 되지 않는 문제가 있습니다.

derrickkim0109 replied:
isUserInteractionEnabled는 탭 제스처로 선언 시에 true하는 거 말고 다른 부분이 있을까요? 조금 더 구체적으로 설명해주실 수 있을까요? UIButton이 아닌 어떤 뷰에서 그런걸까요?

inwoodev replied:
일단 [공식문서](https://developer.apple.com/documentation/uikit/uiview/1622577-isuserinteractionenabled) 에 의하면 UIView 같은 경우 디폴트로 `isUserInteractionEnabled` 설정이 true가 되게 됩니다. UIImageView, StackView 그리고 flexContainer(UIView) 모두 UIControl을 상속 받지 않기 때문에 해당 설정을 별도로 true 설정을 해야 버튼의 제스처를 빼앗아 무시하는 불상사를 해결할 . 수있습니다.

imageView 그리고 additionalView에다가 설정을 할 수도 있지만 일단 flexContainer에 적용했을 때 두 view 모두 터치하는데 문제가 없는 것을 확인하고 `isUserInteractionEnabled`설정을 flexContainer에만 적용하여 중복으로 설정을 하는 것을 방지하였습니다!

inwoodev commented on Projects/Shared/DesignSystem/Sources/Buttons/FeelinConfirmButton.swift:
```diff
@@ -0,0 +1,54 @@
+        initialEnabled: Bool = false,
+        title: String,
+        enabledBackgroundColor: UIColor = Colors.active,
+        disabledBackgroundColor: UIColor = Colors.disabled
+    ) {
+        super.init(frame: .zero)
+        self.isEnabled = initialEnabled
+        self.enabledBackgroundColor = enabledBackgroundColor
+        self.disabledBackgroundColor = disabledBackgroundColor
+        setupButton(title: title)
+    }
+    
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+    
+    private func setupButton(title: String) {
+        updateAppearance()
+        
+        self.setTitle(title, for: .normal)
+        self.titleLabel?.font = UIFont.pretendard(size: 16, type: .semiBold)
+    }
+    
+    private func updateAppearance() {
+        if isEnabled {
+            backgroundColor = enabledBackgroundColor
+            setTitleColor(.white, for: .normal)
+        } else {
+            backgroundColor = disabledBackgroundColor
+            setTitleColor(.white, for: .normal)
+        }
+    }
+    
+    public override var isEnabled: Bool {
+        didSet {
+            updateAppearance()
+        }
+    }
+}
```
flexLayout을 활용할 경우 NSLayoutConstraint과 같은 오토레이아웃 설정을 추가할 경우 레이아웃 문제가 발생합니다. 따라서 공용 컴포넌트인 FeelinConfirmButton에도 오토레이아웃 설정을 통해 height 및 width 설정은 하지 않았습니다.

inwoodev commented on Projects/Feature/Onboarding/Interface/Sources/ViewControllers/UseAgreementViewController.swift:
```diff
@@ -0,0 +1,99 @@
+
+public final class UseAgreementViewController: UIViewController {
+    // MARK: - View
+    private var agreementTitleLabel: UILabel = {
+        let label = UILabel()
+        label.numberOfLines = 0
+        label.text = "Feelin 이용을 위해 약관을\n동의해주세요"
+        label.font = SharedDesignSystemFontFamily.Pretendard.bold.font(size: 24)
+        label.textColor = Colors.gray08
+        
+        return label
+    }()
+    private let useAgreementListView: UseAgreementListView = .init()
```
이런 원시값들은 추후 tuist로 관리해야 합니다. 이슈로 추가할게요

derrickkim0109 replied:
**P3**
개행이 필요해 보여요.

inwoodev replied:
개행은 일단 적용할게요. 이 부분도 코드 컨벤션에 적용할 수 있다면 적용해야 겠네요~:)

inwoodev commented on Projects/Feature/Onboarding/Interface/Sources/ViewControllers/UseAgreementViewController.swift:
```diff
@@ -0,0 +1,99 @@
+    }
+}
+
+// MARK: - Agreement Buttons
+private extension UseAgreementViewController {
+    var allAgreeButton: UIButton {
+        return useAgreementListView.allAgreeButton
+    }
+    
+    var ageAgreeButton: UIButton {
+        return useAgreementListView.ageAgreeButton
+    }
+    
+    var serviceUsageAgreeButton: UIButton {
+        return useAgreementListView.serviceUsageAgreeButton
+    }
+    
+    var checkServiceUsageButton: UIButton {
+        return useAgreementListView.checkServiceUsageButton
+    }
+    
+    var personalInfoAgreeButton: UIButton {
+        return useAgreementListView.personalInfoAgreeButton
+    }
+    
+    var checkPersonalInfoUsageButton: UIButton {
+        return useAgreementListView.checkPersonalInfoUsageButton
+    }
+}
```
추후 뷰모델과 바인딩시 코드 길이를 줄이기 위해서 private extension으로 빼두었습니다.

inwoodev commented on Projects/Feature/Onboarding/Interface/Sources/Components/UserAgreementListView.swift:
```diff
@@ -0,0 +1,199 @@
+        return label
+    }()
+    
+    lazy var allAgreeButton: UIButton = {
+        let subFlexContainer = UIView()
+        
+        subFlexContainer.flex.direction(.row).define { flex in
+            flex.addItem(self.allAgreeLabel)
+                .marginLeft(12)
+        }
+        
+        let button = CheckBoxButton(additionalView: subFlexContainer)
+        button.configuration = UIButton.Configuration.bordered()
+        
+        button.configurationUpdateHandler = { [weak self] button in
+            switch button.state {
+            case .normal:
+                button.configuration?.baseForegroundColor = Colors.gray04
+                button.configuration?.baseBackgroundColor = Colors.background
+                self?.allAgreeLabel.textColor = Colors.gray04
+            case .selected:
+                button.configuration?.baseBackgroundColor = Colors.alertSuccess
+                self?.allAgreeLabel.textColor = Colors.alertSuccess
+                
+            default:
+                return
+            }
+        }
+
+        return button
+    }()
```
전체동의 버튼같은 경우 다른 동의 버튼과 다르게 선택될 경우 배경색상을 변경해야 합니다.
해당 설정을 위와 같이 configurationHandler를 활용하여 구현 해 봤습니다.

inwoodev commented on Projects/Feature/Onboarding/Interface/Sources/Components/UserAgreementListView.swift:
```diff
@@ -0,0 +1,199 @@
+                flex.addItem()
+                    .direction(.row)
+                    .define { flex in
+                        flex.addItem(personalInfoAgreeButton)
+                            .grow(1)
+                        
+                        flex.addItem(checkPersonalInfoUsageButton)
```
`.grow`를 통해 Spacer와 같이 빈 공간을 채워주는 뷰가 별도로 필요가 없습니다 :) FlexLayout이 이건 정말 잘 만들어 놓았네요.

inwoodev commented on Projects/Feature/Onboarding/Interface/Sources/Components/UserAgreementListView.swift:
```diff
@@ -0,0 +1,199 @@
+        return label
+    }
+    
+    func createMandatoryBorderedView() -> UIView {
+        let subflexContainer = UIView()
+        
+        let label = UILabel()
+        label.text = "필수"
+        label.font = SharedDesignSystemFontFamily.Pretendard.medium.font(size: 14)
+        label.textColor = Colors.gray05
+        label.textAlignment = .center
+        
+        subflexContainer.flex
+            .justifyContent(.center)
+            .alignItems(.center)
+            .addItem()
+            .size(.init(width: 49, height: 28))
+            .cornerRadius(14)
+            .border(2, Colors.gray01)
+            .define { flex in
+                flex.addItem(label)
+                    .marginVertical(4)
+                    .marginHorizontal(12)
+            }
+        
+        return subflexContainer
+    }
```
Z-Stack이 명시되어있지는 않지만 이렇게 컨테이너 자체에 shape을 넣을 수 있어서 이를 활용하여 `필수` UI를 구현 해 보았습니다!

derrickkim0109 replied:
이 부분에 대해서 추가적으로 설명해주실 수 있나요? FlexLayout은 ZStack을 지원안하지 않나요?

inwoodev replied:
단순 컨테이너의 모양 정도는 위와 같이 설정할 수 있음을 의미한 코멘트였습니다. 그럼에도 `addItem()`을 적절히 사용한다면 Z-Stack 또는 `overlay`와 같이 활용 해 볼 수 있겠다는 생각이 문득 드네요(물론 이 부분은 추가적인 삽질이 필요해 보입니다:))

inwoodev commented on Projects/Feature/Onboarding/Interface/Sources/Components/CheckBoxButton.swift:
```diff
@@ -0,0 +1,104 @@
+    
+    private var additionalView: UIView?
+    
+    public init(
+        additionalView: UIView? = nil,
+        action: UIAction? = nil
+    ) {
+        super.init(frame: .zero)
+        if let action = action {
+            self.addAction(action, for: .touchUpInside)
+        }
+        addSubview(flexContainer)
+        self.additionalView = additionalView
+        
+        if let additionalView = additionalView {
+            addSubview(additionalView)
+        }
+        
+        setupButton()
+    }
```
외부에서 뷰를 주입 받은 뒤 flexlayout을 활용하여 trailing 부분에 추가하는 방식으로 CheckboxView를 구현 해 봤습니다.

derrickkim0109 commented on Projects/Feature/Onboarding/Example/Sources/AppDelegate.swift:
```diff
@@ -0,0 +1,36 @@
+@main
+class AppDelegate: UIResponder, UIApplicationDelegate {
+    
+    func application(
+        _ application: UIApplication,
+        didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
+    ) -> Bool {
+        return true
+    }
+    
+    // MARK: UISceneSession Lifecycle
+    
+    func application(
+        _ application: UIApplication,
+        configurationForConnecting connectingSceneSession: UISceneSession,
+        options: UIScene.ConnectionOptions
+    ) -> UISceneConfiguration {
+        return UISceneConfiguration(name: "Default Configuration", sessionRole: connectingSceneSession.role)
+    }
+    
+    func application(
+        _ application: UIApplication,
+        didDiscardSceneSessions sceneSessions: Set<UISceneSession>
+    ) {
+    }
+    
```
AppDelegate, SceneDelegate는 example할 때, 매번 필요한 거죠?

inwoodev replied:
네네, 그냥 조그만 앱이라고 보시면 될 것 같아요. 특히 AppDelegate같은 경우 @main 어노테이션을 통해 앱 자체의 진입점을 지정 해주기 때문에 필수라고 생각됩니다. 

반면에 `SceneDelegate`같은 경우 불필요 할수도 있겠다는 생각이 드네요. 일단 example app 같은 경우 싱글 window scene 기반일 것이기에 없어도 될 듯 하네요. AppDelegate의 info.plist에서 SceneDelegate에 대한 엔트리포인트를 제거하는 것과 같은 추가적인 세팅이 필요하긴 하지만요 :)

derrickkim0109 replied:
이 부분은 기존에 제임스가 설정하시던대로 하면 좋을 것 같아요~

derrickkim0109 commented on Projects/Feature/Onboarding/Interface/Sources/Components/CheckBoxButton.swift:
```diff
@@ -0,0 +1,104 @@
+import SharedDesignSystem
+import UIKit
+
+open class CheckBoxButton: UIButton {
+    let flexContainer = UIView()
```
- open인 이유가 있을까요?
- final이 없는데 이유가 있을까요?

inwoodev replied:
음 일단 open으로 둔 이유는 개발당시 해당 뷰를 오버라이드할 수도 있으면서도 Interface가 아닌 source에 두고 활용하자 하는 막연한 생각에 설정을 해 보았는데요. 일단 현재로서는 따로 상속을 받지 않아도 될듯 하여 final로 수정해도 무방할듯 하며 interface 타겟에 있기에 open도 할 필요가 없어 보이네요. 수정할게요~

derrickkim0109 commented on Projects/Feature/Onboarding/Interface/Sources/Components/CheckBoxButton.swift:
```diff
@@ -0,0 +1,104 @@
+    }
+    
+    required public init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setupButton()
```
**P3**
- 이 부분은 따로 `setupButton`를 선언할 필요가 없지 않나요?
- super.init(coder: coder) 필요 없지 않을까요?
- available(*, unavailable) 선언해주시면 좋을 것 같아요.

inwoodev replied:
좋은 포인트입니다. 반영할게요~

다만 `@available(*, unavailable)`어노테이션을 적용하여도 super.init(coder: coder)를 통해 부모 이니셜라이저를 호출하지 않으면 아래와 같은 오류가 발생하기 때문에 `super.init(coder: coder)`는 호출을 꼭 해주어야 합니다!

`'self' used before 'super.init' call`

derrickkim0109 replied:
super.init(coder: coder) 이걸 안하고 보통 `fatalError()`를 쓰는 거 아닌가요?

제가 말씀드렸던 부분의 요지는 NSCoder를 사용할 필요가 없으니까 fatalError()를 작성하면 되지 않을까였습니다. 

@available(*, unavailable)로 해당 메서드가 사용되지 않는다는 것을 명시했다면 fatalError를 사용해도 되지않을까 생각했습니다.

derrickkim0109 commented on Projects/Feature/Onboarding/Interface/Sources/Components/CheckBoxButton.swift:
```diff
@@ -0,0 +1,104 @@
+        setUpView()
+    }
+    
+    @objc private func buttonTapped() {
+        isChecked.toggle()
+        isSelected.toggle()
+    }
+    
```
@objc로 선언되는 메서드의 네이밍은 Tapped가 뒤에 붙는 방식으로 하는 것을 선호하시나요?
저는 didTap~으로 만들긴했어요.

inwoodev replied:
요것도 저희 주말에 컨벤션 작성할 때 정해보시죠. 일단 컨벤션 정의 전 PR이기에 일단 추가 수정은 하지 않겠습니다~

derrickkim0109 commented on Projects/Feature/Onboarding/Interface/Sources/Components/CheckBoxButton.swift:
```diff
@@ -0,0 +1,104 @@
+        self.checkBoxImageView.image = image
+    }
+    
+    private func setUpView() {
```
**P4**
setup ~이지 않나요?

inwoodev replied:
요 부분도 컨벤션에 추가 할까요? ㅎㅎ. 사실 영문법으로 보았을 때 setup은 명사이며 set up은 동사입니다. 이 부분은 메서드이고 동사를 활용하는게 더 적절해 보이기에 후자가 더 적절한 접근으로 보이며 단어 두 개에 카멜케이스를 적용한다면 setUp이 더 맞지 않나 생각이 드네용 :)

derrickkim0109 replied:
아하ㅎㅎ 그렇군요ㅎ 
제임스 말에 동의합니다~!

derrickkim0109 commented on Projects/Feature/Onboarding/Interface/Sources/Components/UserAgreementListView.swift:
```diff
@@ -0,0 +1,199 @@
+        return button
+    }()
+    
+    lazy var ageAgreeButton: UIButton = {
+        return self.createAgreementButton("만 14세 이상 가입 동의")
+    }()
+    
+    lazy var serviceUsageAgreeButton: UIButton = {
+        return self.createAgreementButton("서비스 이용약관 동의")
+    }()
+    
+    lazy var personalInfoAgreeButton: UIButton = {
+        return self.createAgreementButton("개인정보처리방침 동의")
+    }()
+    
+    lazy var checkServiceUsageButton: UIButton = {
+        let button = self.createCheckUsageButton(title: "보기")
+        return button
+    }()
+    
+    lazy var checkPersonalInfoUsageButton: UIButton = {
+        let button = self.createCheckUsageButton(title: "보기")
+        return button
+    }()
+    
```
- 해당 뷰들이 밖에서 알아야하는 이유가 있을까요?

- lazy var로 선언하신 이유가 있을까요?

inwoodev replied:
저번에 잠깐 얘기를 나눈 것 같이 저는 Combine을 활용하여 해당 버튼들을 `UseAgreementViewController`에서 접근하여 ViewModel과 바인딩시 활용할 수 있도록 internal로 선언을 하였구요.

그리고 위 프로퍼티들이 연산프로퍼티가 아닌 `lazy`인 이유는 해당 버튼들이 호출될 때 마다 매번 새로운 버튼인스턴스가 생성될 필요가 없다고 생각했기 때문입니다. 한번 생성된 후 재사용되는 것이 성능 면에서도 더 유리하다고 생각했기에 위와 같이 구현하였습니다 :)

derrickkim0109 commented on Projects/Feature/Onboarding/Interface/Sources/ViewControllers/UseAgreementViewController.swift:
```diff
@@ -0,0 +1,99 @@
+        }
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
```
여기도 위에서 언급한 내용과 동일합니다.

inwoodev replied:
[코멘트](https://github.com/project-lyrics/app-iOS/pull/12#discussion_r1635201125) 와 동일한 이유로 필요합니다!!

derrickkim0109 commented on Projects/Feature/Onboarding/Interface/Sources/ViewControllers/UseAgreementViewController.swift:
```diff
@@ -0,0 +1,99 @@
+        super.init(coder: coder)
+    }
+    
+    @available(*, unavailable)
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        
+    }
```
viewDidLoad에  @available(*, unavailable)를 선언하신 이유가 있을까요?

inwoodev replied:
부끄럽지만 요건 제 실수입니다 ㅎㅎ; 수정할게요!

derrickkim0109 commented on Projects/Shared/DesignSystem/Sources/Buttons/FeelinConfirmButton.swift:
```diff
@@ -0,0 +1,54 @@
+    }
+    
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
```
@available(*, unavailable)를 선언했다면 super를 안해도 되지 않을까요?

inwoodev replied:
습관적으로 먼저 다 super적용한 뒤 어노테이션을 적용해서 그런듯 하네요. 수정할게요~

inwoodev replied:
[코멘트](https://github.com/project-lyrics/app-iOS/pull/12#discussion_r1635201125)와 동일한 이유로 필요합니다.

derrickkim0109 commented on Projects/Shared/DesignSystem/Sources/Buttons/FeelinConfirmButton.swift:
```diff
@@ -0,0 +1,54 @@
+    }
+    
+    private func updateAppearance() {
+        if isEnabled {
+            backgroundColor = enabledBackgroundColor
+            setTitleColor(.white, for: .normal)
+        } else {
+            backgroundColor = disabledBackgroundColor
+            setTitleColor(.white, for: .normal)
+        }
```
**P3**
삼항연산자를 사용하는 건 어떻게 생각하세요?
```swift
backgroundColor = isEnabled ? enabledBackgroundColor : disabledBackgroundColor
```

inwoodev replied:
좋은 의견입니다. 반영할게요!!

derrickkim0109 reviewed (commented):
수고하셨습니다~!

궁금한 점에 대해 코멘트 남겼습니다~!

derrickkim0109 commented:
@noeyiz 지연님 궁금하신 점이나 말씀하시고 싶은 의견이 있으시다면 남겨주세요~

noeyiz commented:
> @noeyiz 지연님 궁금하신 점이나 말씀하시고 싶은 의견이 있으시다면 남겨주세요~

1. 위에 코멘트 남긴 내용인데, 모듈 임포트 관련 질문입니당. 모듈마다 XXXExport 파일이 있는 것을 확인했고, 이것이 하위 모듈을 따로 임포트하지 않기 위함인 것으로 이해를 했는데요. 코드 상에는 필요한 하위 모듈을 선택하여 임포트가 되어 있더라고요! 그래서 (1) 필요한 하위 모듈을 임포트하면 된다. (2) 추후에 상위 모듈만 임포트하도록 바꿀 것이다. 둘 중에 어떤 것이 맞는지 궁금합니다 ?-?

2. 변수/함수 네이밍이나 임포트 순서 등 컨벤션을 보다 확실하게 정하면 더 좋을 것 같습니다 ! (P4)

inwoodev commented:
`export`도 그렇고 tuist 정리가 분명히 필요해 보이네요. 이번 주말에 같이 모여서 한번 이 부분에 대해서도 논의 하시죠~

inwoodev committed:
refactor : 개행 수정, 접근제어 범위 수정, if문 삼항연산자로 대체, 어노테이션 수정

- 불필요한 @available(*, unavailable) 제거
- 사용되지 않지만 필요한 메서드에 @available(*, unavailable) 어노테이션 추가
- 변수간 일정한 개행 적용
- 삼항연산자를 활용하여 가독성 개선 및 코드 중복 최소화
- 불필요하게 open 및 public설정이 된 객체들 접근제어 범위 수정(internal로)
- 상속이 필요 없는 객체들에게 final 키워드 추가

inwoodev commented:
1차 수정 완료 하였습니다. 추가적으로 수정사항이 보이는 부분이 있다면 가감없이 코멘트 남겨주시면 감사하겠습니다. 머지해도 괜찮아 보인다면 따봉 이모티콘 부탁드립니다 :)

derrickkim0109 commented:
수고하셨습니다~!
